pkg/skaffold/docker/auth.go

```diff
@@ -68,7 +69,25 @@ func (credsHelper) GetAuthConfig(registry string) (types.AuthConfig, error) {
 
 	gcp.AutoConfigureGCRCredentialHelper(cf, registry)
 
-	return cf.GetAuthConfig(registry)
+	authConfig, err := cf.GetAuthConfig(registry)
+
+	if err != nil {
+		return types.AuthConfig{}, errors.Wrap(err, "docker config")
+	}
+
+	return asAPIAuthConfig(authConfig), nil
+}
+
+func asAPIAuthConfig(cliConfig types2.AuthConfig) types.AuthConfig {
```

Comment:
I don't think you need to do this. these `AuthConfig` structs are identical:

https://godoc.org/github.com/docker/cli/cli/config/types#AuthConfig
https://godoc.org/github.com/docker/docker/api/types#AuthConfig

you can just remove the old dependency and return the new one, leaving the code untouched. they should be interchangeable

Reply:
not sure if I understand your thought. 

**ImageBuildOptions** can receive only **AuthConfigs** from api/types (https://github.com/GoogleContainerTools/skaffold/blob/master/pkg/skaffold/docker/image.go#L155) however  **ConfigFile#GetAuthConfig** (https://github.com/slamdev/skaffold/blob/master/pkg/skaffold/docker/auth.go#L72) now returns the one from cli/config/types

could you clarify how this can be solved without converting one struct to another?

Reply:
ahh, you're right. I thought we were using this struct directly but I missed this.